test/system/src/test/java/io/pravega/test/system/AutoScaleTest.java

```diff
@@ -84,16 +81,27 @@ public static void setup() {
         //get the zk ip details and pass it to bk, host, controller
         URI zkUri = zkUris.get(0);
         //2, check if bk is running, otherwise start, get the zk ip
-        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        Service bkService = Utils.createBookkeeperService(zkUri);
         if (!bkService.isRunning()) {
             bkService.start(true);
         }
 
         List<URI> bkUris = bkService.getServiceDetails();
         log.debug("bookkeeper service details: {}", bkUris);
 
+        //start HDFS
+        URI hdfsUri = null;
+        if (Utils.isDockerLocalExecEnabled()) {
```

Comment:
Same comment here about environment specific logic.

Reply:
As we are using a plain cluster, with nothing installed apart from docker daemon, we need to deploy even hdfs  with other pravega components, when we are executing system tests through docker swarm .
In our previous usages, we always have hdfs deployed in the cluster as a part of cluster creation.
So you will seeing this in every test case.

Reply:
Same issue, we need to abstract away any behavior that is specific to the environment and encapsulate it in some helper class.

Reply:
Done, taken care in helper method.